sphinx/writers/latex.py

```diff
@@ -1852,6 +1852,11 @@ def visit_figure(self, node):
             self.body.append('\\begin{wrapfigure}{%s}{%s}\n\\centering' %
                              (node['align'] == 'right' and 'r' or 'l', length or '0pt'))
             self.context.append(ids + '\\end{wrapfigure}\n')
+
+            # emit a warning about wrapfig (refs: #3289)
+            logger.warning(__('A figure having :align: option has detected. '
+                              'LaTeX might not layout it well. Please check generated PDF.'),
```

Comment:
Perhaps:
 
    'LaTeX perhaps did not layout it correctly. Please check generated PDF.'

But checking PDFs is time costly...